editor.planx.uk/src/@planx/components/MultipleFileUpload/Public.tsx

```diff
@@ -65,23 +63,26 @@ function Component(props: Props) {
     const passport = useStore.getState().computePassport();
     const fileList = createFileList({ passport, fileTypes: props.fileTypes });
     setFileList(fileList);
-  }, []);
 
-  useEffect(() => {
-    // TODO: Re-map slots to userfiles also?
-    const recoveredSlots: FileUploadSlot[] = getRecoveredSlots(
-      props.previouslySubmittedData,
-      fileList
-    );
-    setSlots(recoveredSlots);
-  }, [props.previouslySubmittedData, fileList]);
+    if (props.previouslySubmittedData) {
+      const recoverredData = getRecoveredData(
+        props.previouslySubmittedData,
+        fileList
+      );
+      setSlots(recoverredData.slots);
+      setFileList(recoverredData.fileList);
+      setIsUserReturningToNode(true);
```

Comment:
This feels a bit awkward but it works - I couldn't find another way using existing state and stores to track if a user has just clicked back. The modal needs to still behave as normal when uploading files after clicking "back" (e.g. to upload more files)